redirect/redirect_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// Package redirect implements redirection based login.
+package redirect_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	errgo "gopkg.in/errgo.v1"
+	"gopkg.in/macaroon-bakery.v2/httpbakery"
+
+	"gopkg.in/CanonicalLtd/candidclient.v1/params"
+	"gopkg.in/CanonicalLtd/candidclient.v1/redirect"
+)
+
+func TestRedirectURL(t *testing.T) {
+	c := qt.New(t)
+
+	info := redirect.InteractionInfo{
+		LoginURL: "https://www.example.com/login",
+	}
+	rurl := info.RedirectURL("https://www.example.com/callback", "12345")
+	c.Assert(rurl, qt.Equals, "https://www.example.com/login?return_to=https%3A%2F%2Fwww.example.com%2Fcallback&state=12345")
+
+	info = redirect.InteractionInfo{
+		LoginURL: "https://www.example.com/login?domain=test",
+	}
+	rurl = info.RedirectURL("https://www.example.com/callback", "12345")
+	c.Assert(rurl, qt.Equals, "https://www.example.com/login?domain=test&return_to=https%3A%2F%2Fwww.example.com%2Fcallback&state=12345")
+}
+
+func TestInteractor(t *testing.T) {
+	c := qt.New(t)
+	ctx := context.Background()
+
+	var i redirect.Interactor
+	c.Assert(i.Kind(), qt.Equals, redirect.Kind)
+
+	req, err := http.NewRequest("GET", "https://www.example.com/discharge", nil)
+	c.Assert(err, qt.Equals, nil)
+	irerr := httpbakery.NewInteractionRequiredError(nil, req)
+	// Fake an empty InteractionRequiredError
+	irerr.Info = &httpbakery.ErrorInfo{}
+
+	_, err = i.Interact(ctx, nil, "", irerr)
+	c.Assert(errgo.Cause(err), qt.Equals, httpbakery.ErrInteractionMethodNotFound)
+
+	redirect.SetInteraction(irerr, "https://www.example.com/login", "https://www.example.com/token")
+
+	_, err = i.Interact(ctx, nil, "", irerr)
+	c.Assert(httpbakery.IsInteractionError(err), qt.Equals, true, qt.Commentf("%s", err))
```

Comment:
Replace this with `c.Assert(err, qt.Satisfies, httpbakery.IsInteractionError)` ?

Reply:
done